### ISSUE TITLE
phpseclib sshconnect login failed.

### ISSUE THREAD

**Issue by github718** [phpseclib/phpseclib#1484]
I made an SSH connection using phpseclib but cannot log in.
define('NET_SSH2_LOGGING', NET_SSH2_LOG_COMPLEX);
I used $ssh->getLastError(); and $ssh->getErrors(); but there are no errors.
This is connect used code.
>include(__DIR__.'/Net/SSH2.php');
>define('NET_SSH2_LOGGING', NET_SSH2_LOG_COMPLEX);
>$ssh = new Net_SSH2('hostname',portnum);
>if (!$ssh->login('username','password')) {
>     echo $ssh->getLog();     
>     //exit('login failed');
>}

This is the log I got with $ssh->getLog();
[phpseclib.log](https://github.com/phpseclib/phpseclib/files/4828563/phpseclib.log)

And this is the SSH connection PuTTY log, 
[putty.log](https://github.com/phpseclib/phpseclib/files/4828555/putty.log)

How can phpseclib I log in?

**Comment by github718** [comment]
> I made an SSH connection using phpseclib but cannot log in.
> define('NET_SSH2_LOGGING', NET_SSH2_LOG_COMPLEX);
> I used $ssh->getLastError(); and $ssh->getErrors(); but there are no errors.
> This is the log I got with $ssh->getLog();
> [phpseclib.log](https://github.com/phpseclib/phpseclib/files/4828563/phpseclib.log)
> 
> And this is the SSH connection PuTTY log,
> [putty.log](https://github.com/phpseclib/phpseclib/files/4828555/putty.log)
> 
> How can phpseclib I log in?

**Comment by terrafrost** [comment]
The issue is the signature verification algorithms the server supports - it only supports ecdsa-sha2-nistp256 and ssh-ed25516. phpseclib v3 supports those algorithms but not v1 or v2.

I'm still working on the documentation for v3 and haven't made an official release of it yet. You're free to use that branch, however. I can provide you with any help that you might need to transition over!

**Comment by github718** [comment]
Thank you for your reply terrafrost.
I am using Openssh server on Ubuntu 16.04.6 LTS server OS, do you know the settings to allow to connect with v2?
If you can't connect without v3, you need help.

**Comment by terrafrost** [comment]
It won't work with v2, based on your log file. BUT if you install phpseclib with Composer you can install v3 thusly:

```
composer require phpseclib/phpseclib:3.0.x-dev
```
Once you do that here's what you'd do:

```php
use phpseclib3\Net\SSH2;

$ssh = new SSH2('hostname',portnum);
$ssh->login('username', 'password');
```
The API for SSH2 in v3 is pretty much the same as it is for v2. It works a bit differently with public key auth but you don't appear to be doing that!

**Comment by github718** [comment]
Thank you for your reply.
I reinstalled Openssh to initialize the server settings.
so,now I can connect even with v2.
I thank you very much.
However, which is more stable, v2 or v3?
If there stable  particular, I will try installing v3 you taught me

**Comment by terrafrost** [comment]
> However, which is more stable, v2 or v3?

v2 has been pretty heavily used for 5 years. Because v3 hasn't yet had an official release it doesn't really have any heavy usage yet.

TLDR I'd say v2.

**Comment by github718** [comment]
Thank you for teaching.terrafrost.
>TLDR I'd say v2.
Then I will move it with v2.
